### PR TITLE
Update to KDE runtime 5.12 and build from master

### PR DESCRIPTION
**Pull request by RogueScholar**
This is my first successful attempt to modify an existing flatpak manifest which builds without error on my platform. I've tried to adhere to the documentation as closely as possible but expect that there's still much room for improvement. Any advice or constructive criticism is most welcome.

Signed-off-by: Peter J. Mello <admin@petermello.net>

### CODE REVIEW

**Comment by RogueScholar** [comment]
Yippee!!! A thousand pardons to the poor soul(s) that may eventually be forced to sift through my commit log on this one; the only way to describe it is as something akin to going fishing armed with only a shotgun and a bag of shells... :fish: :gun:

I'm still intent on demystifying what is preventing the qTox build process itself from recognizing and linking to the toxav and toxencryptsave modules resident within the toxcore library as that puts a damper on the voice and video chat functions of the app. Once again, any thoughts or suggestions are most welcome.

**Comment by TingPing** [comment]
`config.guess` and `config.sub` are in the runtime, downloading it is overkill.

File: io.github.qtox.qTox.json [hunk 1]
```diff
@@ -22,49 +22,87 @@
         "--device=all"
     ],
     "build-options": {
-        "cflags": "-O3 -DSQLITE_HAS_CODEC",
-        "cxxflags": "-O3"
+        "cflags": "-O3 -fPIC",
```
Comment (TingPing) [inline io.github.qtox.qTox.json]: These should probably be removed, the runtime now sets default cflags.

File: io.github.qtox.qTox.json [hunk 2]
```diff
@@ -22,49 +22,87 @@
...
+        "cxxflags": "-O3 -fPIC"
     },
     "cleanup": [
         "/include",
         "/lib/pkgconfig",
         "/share/man"
     ],
     "modules": [
+		"shared-modules/SDL/SDL-1.2.15.json",
+		"shared-modules/gtk2/gtk2.json",
```
Comment (TingPing) [inline io.github.qtox.qTox.json]: What uses gtk2?

File: io.github.qtox.qTox.json [hunk 3]
```diff
@@ -22,49 +22,87 @@
...
                     "type": "archive",
-                    "url": "https://linuxtv.org/downloads/v4l-utils/v4l-utils-1.14.2.tar.bz2",
-                    "sha256" : "e6b962c4b1253cf852c31da13fd6b5bb7cbe5aa9e182881aec55123bae680692"
+                    "url": "https://linuxtv.org/downloads/v4l-utils/v4l-utils-1.16.3.tar.bz2",
+                    "sha256": "7c5c0d49c130cf65d384f28e9f3a53c5f7d17bf18740c48c40810e0fbbed5b54"
                 }
             ]
         },
+        {
+        	"name": "opus",
```
Comment (TingPing) [inline io.github.qtox.qTox.json]: This is in the runtime.

File: io.github.qtox.qTox.json [hunk 4]
```diff
@@ -22,49 +22,87 @@
...
+        	"sources": [
+        		{
+        			"type": "archive",
+        			"url": "https://archive.mozilla.org/pub/opus/opus-1.3.tar.gz",
+        			"sha256": "4f3d69aefdf2dbaf9825408e452a8a414ffc60494c70633560700398820dc550"
+        		}
+        	]
+        },
+        {
+        	"name": "vpx",
```
Comment (TingPing) [inline io.github.qtox.qTox.json]: This is in the runtime.

File: io.github.qtox.qTox.json [hunk 5]
```diff
@@ -80,25 +118,83 @@
...
+        	"sources": [
+        		{
+        			"type": "archive",
+        			"url": "https://github.com/irungentoo/filter_audio/archive/v0.0.1.tar.gz",
+        			"sha256": "fb135592c5133c3b4b664da18f988f58609db912f204059abe16277df044a366"
+        		}
+        	]
+        },
+        {
+        	"name": "openal",
```
Comment (TingPing) [inline io.github.qtox.qTox.json]: This is in the runtime.

File: io.github.qtox.qTox.yml [hunk 6]
```diff
@@ -0,0 +1,145 @@
+---
```
Comment (TingPing) [inline io.github.qtox.qTox.yml]: Why is this here twice, just pick a format.

File: io.github.qtox.qTox.json (hunk 2 above)
Comment (anthonybilinski) [reply io.github.qtox.qTox.json]: In v1.16.3 gtk2 was used by one of our status icon implementations. It has been removed on master since, so won't be a dependency in the next release.
